rle_array/_slicing.py

```diff
@@ -0,0 +1,395 @@
+            raise TypeError(f"stop must be int but is {type(stop).__name__}")
+        if not isinstance(step, int):
+            raise TypeError(f"step must be int but is {type(step).__name__}")
+        if not isinstance(container_length, int):
+            raise TypeError(
+                f"container_length must be int but is {type(container_length).__name__}"
+            )
+
+        self._start = start
+        self._stop = stop
+        self._step = step
+        self._container_length = container_length
+
+        self._verify()
+
+    def _verify(self) -> None:
+        """
+        Verify integrity.
+        """
+        if self.container_length < 0:
+            raise ValueError(
+                f"container_length ({self.container_length}) must be greater or equal to zero"
+            )
+        elif self.container_length == 0:
+            self._verify_container_empty()
+        else:
+            self._verify_container_not_empty()
+
+    def _verify_container_empty(self) -> None:
+        """
+        Verify integrity in case the container is empty (``container_length = 0``).
+        """
+        # empty container => special values required
+        if self.start != 0:
+            raise ValueError(
+                f"for empty containers, start must be 0 but is {self.start}"
+            )
+
+        if self.stop != 0:
+            raise ValueError(f"for empty containers, stop must be 0 but is {self.stop}")
+
+        if self.step != 1:
+            raise ValueError(f"for empty containers, step must be 1 but is {self.step}")
+
+    def _verify_container_not_empty(self) -> None:
+        """
+        Verify integrity in case the container is not empty (``container_length > 0``).
+        """
+        if (self.start < 0) or (self.start >= self.container_length):
+            raise ValueError(
+                f"start ({self.start}) must be in [0,{self.container_length}) but is not"
+            )
+
+        if (self.stop < -abs(self.step)) or (
+            self.stop >= self.container_length + abs(self.step)
+        ):
+            raise ValueError(
+                f"stop ({self.stop}) must be in [{-abs(self.step)},{self.container_length + abs(self.step)}) but is not"
+            )
+
+        if self.start == self.stop:
+            # empty slice
+            if self.start != 0:
+                raise ValueError(
+                    f"for empty slices, start and stop must be 0 but are {self.start}"
+                )
+            if self.step != 1:
+                raise ValueError(f"for empty slices, step must be 1 but is {self.step}")
+        else:
+            # non-empty slice
+            if self.step == 0:
+                raise ValueError("step cannot be zero")
+            elif self.step > 0:
+                # forward slice
+                if self.start > self.stop:
+                    raise ValueError(
+                        "for forward slices, stop must be greater or equal to start"
```

Comment:
You say must be greater or equal but just check if they are greater here

Reply:
both are true :D Thing is: if they are equal, it is an empty slice and we'll never get to this point.